pyproject.toml

```diff
@@ -1,42 +1,54 @@
-[mypy]
-python_version = 3.9
-show_error_codes = true
-follow_imports = silent
-ignore_missing_imports = true
-strict_equality = true
-warn_incomplete_stub = true
-warn_redundant_casts = true
-warn_unused_configs = true
-warn_unused_ignores = true
-check_untyped_defs = true
-disallow_incomplete_defs = true
-disallow_subclassing_any = true
-disallow_untyped_calls = true
-disallow_untyped_decorators = true
-disallow_untyped_defs = true
-no_implicit_optional = true
-warn_return_any = true
-warn_unreachable = true
+[build-system]
+requires = ["setuptools~=62.3", "wheel~=0.37.1"]
+build-backend = "setuptools.build_meta"
 
-[mypy-arelle.*]
+[tool.mypy]
 check_untyped_defs = false
+
 disallow_incomplete_defs = false
 disallow_subclassing_any = false
 disallow_untyped_calls = false
 disallow_untyped_decorators = false
 disallow_untyped_defs = false
+
+# Warn when a # type: ignore comment does not specify any error codes
+enable_error_code = "ignore-without-code"
+
+follow_imports = "silent"
+
+ignore_missing_imports = true
```

Comment:
While I'm no expert in mypy, my idea here was to have a very lax config and then enforce strict typing file by file and then, when all files in a folder are strictly typed, add the entire folder to the strict list instead of each individual file. I did follow a similar path for another open source library where I'm one of the maintainers and found it to work quite well.

If we use `strict=true` the help command tells us that the following is enabled so it should suffice to set that for the last part of the config file:
```
Strict mode; enables the following flags: --warn-unused-configs, --disallow-any-
                            generics, --disallow-subclassing-any, --disallow-untyped-calls, --disallow-untyped-
                            defs, --disallow-incomplete-defs, --check-untyped-defs, --disallow-untyped-decorators,
                            --no-implicit-optional, --warn-redundant-casts, --warn-unused-ignores, --warn-return-
                            any, --no-implicit-reexport, --strict-equality, --strict-concatenate
```

Reply:
My understanding of the config syntax is that the first part is global and applies everywhere. Then we filter on all files in the arelle.* space where we disabled all errors as there would be thousands of them. Finally, we turn on strict mode on certain files.

Reply:
Thanks for the clarification. I made the same changes to the config to see what you mean and I agree that a simpler config is better. Do you want me to commit these changes?